packages/database/src/api/Query.ts

```diff
@@ -572,7 +575,28 @@ export class Query {
     Query.validateQueryEndpoints_(newParams);
     if (this.queryParams_.hasEnd()) {
       throw new Error(
-        'Query.endAt: Ending point was already set (by another call to endAt or ' +
+        'Query.endAt: Ending point was already set (by another call to endAt, endBefore, or ' +
+          'equalTo).'
+      );
+    }
+
+    return new Query(this.repo, this.path, newParams, this.orderByCalled_);
+  }
+
+  endBefore(
+    value: number | string | boolean | null = null,
+    name?: string | null
+  ): Query {
+    validateArgCount('Query.endBefore', 0, 2, arguments.length);
+    validateFirebaseDataArg('Query.endBefore', 1, value, this.path, true);
```

Comment:
This is not optional. Do `null` values still work if you pass `false` as the last argument?